client/chat/channel-info-card.tsx

```diff
@@ -1,58 +1,114 @@
 import React, { useEffect, useState } from 'react'
 import styled from 'styled-components'
 import { ChatServiceErrorCode, SbChannelId } from '../../common/chat'
-import ChannelIcon from '../icons/material/image-24px.svg'
 import { MaterialIcon } from '../icons/material/material-icon'
 import { RaisedButton } from '../material/button'
 import Card from '../material/card'
+import { shadow2dp } from '../material/shadows'
 import { isFetchError } from '../network/fetch-errors'
 import { LoadingDotsArea } from '../progress/dots'
 import { useAppDispatch, useAppSelector } from '../redux-hooks'
 import { useStableCallback } from '../state-hooks'
-import { colorError } from '../styles/colors'
-import { body1, Body1, Headline6 } from '../styles/typography'
-import { getChannelInfo, joinChannel, navigateToChannel } from './action-creators'
-
-const Container = styled(Card)`
-  width: 100%;
-  max-width: 440px;
-  min-width: 220px;
-  padding: 16px;
+import { colorTextFaint } from '../styles/colors'
+import { FlexSpacer } from '../styles/flex-spacer'
+import { body1, caption, headline6 } from '../styles/typography'
+import { getBatchChannelInfo, joinChannel, navigateToChannel } from './action-creators'
+import { ChannelBadge } from './channel-badge'
+import { ChannelBanner, ChannelBannerPlaceholderImage } from './channel-banner'
+
+const ChannelCardRoot = styled(Card)`
+  position: relative;
+  width: 352px;
+  padding: 0;
 
   display: flex;
-  flex-direction: row;
-  align-items: center;
+  flex-direction: column;
+
+  contain: content;
 `
 
 const StyledLoadingDotsArea = styled(LoadingDotsArea)`
   width: 100%;
 `
 
-const StyledChannelIcon = styled.svg`
-  width: 56px;
-  height: auto;
-  flex-shrink: 0;
+const ChannelBannerAndBadge = styled.div`
+  box-sizing: content-box;
+  position: relative;
+  padding-bottom: 20px;
+`
+
+const ChannelCardBadge = styled.div`
+  ${shadow2dp};
+  position: absolute;
+  left: 12px;
+  bottom: 0;
+  width: 52px;
+  height: 52px;
+  padding: 6px;
+
+  background: var(--sb-color-background);
+  border-radius: 9999px;
+`
+
+const ChannelName = styled.div`
+  ${headline6};
+  margin-top: 4px;
+  padding: 0 16px;
+`
+
+const ChannelUserCount = styled.div`
+  ${caption};
+  padding: 0 16px;
+`
+
+const ChannelDescription = styled.div`
+  ${body1};
+  margin-top: 16px;
+  padding: 0 16px;
+
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  line-clamp: 3;
+  -webkit-line-clamp: 3;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+const PrivateChannelIcon = styled(MaterialIcon).attrs({ icon: 'lock' })`
+  vertical-align: bottom;
```

Comment:
I'm not totally sure what visual treatment you're trying to go for here, but I struggle to imagine a case where "bottom" is the correct value. Seems more likely you want `text-top` or `baseline`?

Reply:
And whatever value you choose here, you should look how it behaves when the text it is beside extends across multiple lines, because I think this will surprise you

Reply:
Hmm, that's what I thought to. But for some reason it only looked decent to me with `vertical-align: bottom`. Here are some screenshots to show what I mean.

`vertical-align: bottom`:
![image](https://user-images.githubusercontent.com/4757944/228216086-0b3a3549-0303-4ee0-8475-0475682a8770.png)

`vertical-align: baseline`:
![image](https://user-images.githubusercontent.com/4757944/228216188-3d21ded8-c831-4a0e-a7ca-52e24bd8f13e.png)

`vertical-align: text-top`:
![image](https://user-images.githubusercontent.com/4757944/228216300-a3e6c33f-f2c4-4e81-ba05-e93051d8d601.png)

Reply:
I think the issue is that I'm trying to fit a 24px icon in a same line with a text which has a 20px line-height. But even when I increase the combined line-height to 24px, I feel like `vertical-align: bottom` still looks the best for some reason.

Reply:
You could just use a smaller icon, it is very easy to pick one that matches your line height now :) But I don't think that looks good regardless, having the text flow around the icon is very much not what we want. I think it would probably look better to use a larger icon, center it, and put this text below.

If that for some reason doesn't work out, you'd probably just want to use a flex layout and put the icon and the text in separate blocks.

Reply:
Yeah, I think using centered text for it might look better, but otherwise fine, The padding between the icon and the channel name also looks a tad large?